test/io/ferguson/writef.good

```diff
@@ -52,8 +52,19 @@ Part 8
 |7      |
 |7      |
 | 7     |
+|      7|
+|0000007|
+|      7|
+|   7   |
+|0007   |
```

Comment:
Should this have the 0 on both sides?  Seems a bit wrong for the padding character to only apply to one side when center-justified, but if that's how Python behaves then I'm okay with it

Reply:
I initially had zeros on both sides, but then thought that it sort of conflicted with our rule for right-padding with zeros. Where the current documentation for `%-` says:
```
left-justify the converted value instead of right-justifying. Note, if both 0 and - are given, the effect is as if only - were given.
```
I assume that this rule is in place so that the meaning of `5` isn't accidentally changed to `500000` when someone uses `%0<6i`. So I decided to do the analogous thing for `%0^6i`. This does not apply for other padding characters. In those cases it is shown on both sides.

FWIW, it looks like Python and Rust both show zeros on both sides when you center justify a number with `0` as the padding character. So this does not align with what other languages have done.

Let me know what you think.

Reply:
Ah, okay.  In that case, I'd maybe use a different character to demonstrate the padding, or have an additional section demonstrating that 0 is the exception here.  And probably call the behavior out on the documentation for `%^`

Reply:
Sounds good, I'll add another section to that test. And it looks like I have a small typo in the docs for `^`, but they should read:
```
center-justify the converted value instead of right-justifying. Note, if both 0 and ^ are given, zero-padding will only be applied to the left of the numerical value
```

Reply:
Sounds good, thanks!